mdpow/workflows/dihedrals.py

```diff
@@ -252,15 +281,29 @@ def dihedral_groups(dirname, resname, SMARTS=SMARTS_DEFAULT):
 
     '''
 
-    u = build_universe(dirname=dirname)
-    solute = rdkit_conversion(u=u, resname=resname)[1]
-    atom_group_indices = dihedral_indices(dirname=dirname, resname=resname, SMARTS=SMARTS)
-    dihedral_groups = [solute.atoms[list(dihedral_indices)].names for dihedral_indices
-                       in atom_group_indices]
+    dihedral_groups = [solute.atoms[list(a_ind)].names for a_ind
+                       in atom_indices]
 
     return dihedral_groups
 
-def dihedral_groups_ensemble(dirname, atom_group_indices,
+def get_paired_indices(atom_indices, bond_indices, dihedral_groups):
+
+    dg_list = []
+    for dg in dihedral_groups:
+        group = (f'{dg[0]}-{dg[1]}-{dg[2]}-{dg[3]}')
+        dg_list.append(group)
+    all_dgs = tuple(dg_list)
```

Comment:
Why are you making the list a tuple?

Why not just build it with a list comprehension (... or tuple comprehension if you really want a tuple)?
```suggestion
    all_dgs = [f'{dg[0]}-{dg[1]}-{dg[2]}-{dg[3]}' for dg in dihedral_groups]
```

Reply:
I was using tuples during development so that it could not accidentally be modified downstream and because these values should not ever be changed externally, but your method is much cleaner so I will add the changes, thank you.